src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -9849,6 +9849,7 @@ private static Precedence GetPrecedence(SyntaxKind op)
                 case SyntaxKind.PostDecrementExpression:
                 case SyntaxKind.PostIncrementExpression:
                 case SyntaxKind.PredefinedType:
+                case SyntaxKind.QualifiedName:
```

Comment:
>case SyntaxKind.QualifiedName: [](start = 16, length = 30)

It looks like we are getting here only when AliasQualifiedName is used in code. Could you elaborate why that requires special handling for QualifiedName, but doesn't require it for the case when alias qualification isn't used?

Reply:
this does feel weird to me.  i can't see how you'd have an expression where you need to compute precedence where you'd have a QualifiedName. (i can see how an ALiasQualifiedName woudl cause this).

Reply:
The QualifiedName `::A.B` contains an AliasQualifedName `::A` as the left part.

Reply:
why is `::A.B` not a MemberAccessExpressoin with an AliasQuaifiedName as the left part?

Reply:
> why is ::A.B not a MemberAccessExpressoin with an AliasQuaifiedName as the left part?

`ParseQualifiedNameRight(options, left, separator)` calls `_syntaxFactory.QualifiedName(left, separator, right)` when `separator.Kind == SyntaxKind.DotToken`.

Reply:
I guess i don't understand why ParseQualifiedNameRight is being called here.  If we're in an expression context, why would we ever call ParseQualifiedName?  The order, as i understand it  would be that in `A::B.C<D>` we would see the `A`, then the `::`, this would cause us to do ParseAliasQualifiedName.  That will see we have a `::` so we should just consume the `B` and make an AliasQualifiedName.  We'll then pop back up in ParseTerm and call ParsePostFixExpression, which will see the dot and convert the following to MemberAccessExpression